Tests/MoyaTests/TestHelpers.swift

```diff
@@ -100,16 +100,22 @@ enum HTTPBin: TargetType, AccessTokenAuthorizable {
 
     var task: Task {
         switch self {
-        case .basicAuth, .post, .bearer:
-            return .requestParameters(parameters: [:], encoding: URLEncoding.default)
+        case .basicAuth, .bearer:
+            return .request(queryParams: .init(["": ""]))
```

Comment:
If we make QueryParam conform to `ExpressibleByDictionaryLiteral` we can simplify this to `.request(queryParams: ["": ""])`

Reply:
That is a great idea! I'll update it.

Reply:
I'm struggling with the fact that to be encodable, the dict should have its keys `Hashable & Encodable`. I can't express it with `public typealias Key = Encodable & Hashable` because it raises the error `Protocol 'Hashable' can only be used as a generic constraint because it has Self or associated type requirements` when implementing the init.
The only way I see is to have the new init being like `public init<Key: Encodable & Hashable, Value: Encodable>(dictionaryLiteral elements: (Key, Value)...)` but it's not the same definition from `ExpressibleByDictionaryLiteral`.

Reply:
Ohh, I thought it was [String: String] ... Sad :(
But being a constrained generic type I'm not sure you can express that in swift with type alias. 
The only other way you could express that would be if the class itself was generic ... 
```swift
class QueryParam<Key, Value>: ExpressibleByDictionaryLiteral where Key: Codable & Hashable, Value: Codable {
  var d: [Key: Value] = [:]
  required convenience init(dictionaryLiteral elements: (Key, Value)...) {
    self.init()
    self.d = Dictionary(uniqueKeysWithValues: elements)
  }
}
``` 
But that will be harder to the user I think. 

One question tho is if it makes sense for this query param to be generic?
because I'm really not sure a use case where someone would use this 
with another thing other than `[String: String]`...

Reply:
We coud restraint the key to be a `String` and let the Value be an `Encodable` as it does seems to be an issue with `ExpressibleByDictionaryLiteral`. But I fear it closes some possibilities just for the sake of being `ExpressibleByDictionaryLiteral`.
About habing de Generic `QueryParam`, it would allow `ExpressibleByDictionaryLiteral` but then if we provide pairs which type can not be inferred it will quickly be cumbersome to had the generic type to the QueryParam. 
Imo either we find a way to implement `ExpressibleByDictionaryLiteral` without generics or we don't.